sshd-common/src/main/java/org/apache/sshd/common/util/io/DirectoryScanner.java

```diff
@@ -188,21 +187,36 @@ public File getBasedir() {
      *
      * <p>When a pattern ends with a '/' or '\', "**" is appended.</p>
      *
-     * @param includes A list of include patterns.
-     *                 May be {@code null}, indicating that all files
-     *                 should be included. If a non-{@code null}
-     *                 list is given, all elements must be
-     *                 non-{@code null}.
+     * @param includes A list of include patterns. May be {@code null}, indicating
+     * that all files should be included. If a non-{@code null} list is given, all
+     * elements must be non-{@code null}.
      */
-    public void setIncludes(String[] includes) {
-        if (includes == null) {
-            this.includes = null;
-        } else {
-            this.includes = new String[includes.length];
-            for (int i = 0; i < includes.length; i++) {
-                this.includes[i] = normalizePattern(includes[i]);
-            }
-        }
+    public void setIncludes(String... includes) {
+        setIncludes(GenericUtils.isEmpty(includes) ? Collections.emptyList() : Arrays.asList(includes));
+    }
+
+    /**
+     * @return Un-modifiable list of the inclusion patterns
+     */
+    public List<String> getIncludes() {
+        return includePatterns;
```

Comment:
The comment above effectively says "returns an unmodifiable list". But if somebody casts the returned list to an `ArrayList`, it would be modifiable, right? Perhaps a [`Collections.unmodifiableList`](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#unmodifiableList(java.util.List)) is in order?

Reply:
Look closely at the code an you will see the `includePatterns` is **always** unmodifiable...

Reply:
Gotcha. Thanks.